src/components/core/base/BaseVisualization.tsx

```diff
@@ -6,39 +6,30 @@ import {
 } from './VisualizationLoadingHOC';
 import { ILoadingProps } from '../../simple/LoadingComponent';
 import { IErrorProps } from '../../simple/ErrorComponent';
+import { generateErrorMap, IErrorMap } from '../../../helpers/errorHandlers';
 
 export abstract class BaseVisualization<P extends ICommonVisualizationProps & ILoadingInjectedProps, S>
 extends React.Component<P, S> {
+    private errorMap: IErrorMap;
+
+    constructor(props: P) {
+        super(props);
+
+        this.errorMap = generateErrorMap(props.intl);
+    }
 
     public render(): JSX.Element {
         const {
             execution,
             error,
-            isLoading,
-            intl
+            isLoading
         } = this.props;
 
         const ErrorComponent = this.props.ErrorComponent as React.ComponentType<IErrorProps>;
         const LoadingComponent = this.props.LoadingComponent as React.ComponentType<ILoadingProps>;
 
         if (error) {
-            const errorMap = {
-                [ErrorStates.DATA_TOO_LARGE_TO_DISPLAY]: {
-                    icon: 'icon-cloud-rain',
-                    message: intl.formatMessage({ id: 'visualization.ErrorMessageDataTooLarge' }),
-                    description: intl.formatMessage({ id: 'visualization.ErrorDescriptionDataTooLarge' })
-                },
-                [ErrorStates.NO_DATA]: {
-                    icon: 'icon-filter',
-                    message: intl.formatMessage({ id: 'visualization.ErrorMessageNoData' }),
-                    description: intl.formatMessage({ id: 'visualization.ErrorDescriptionNoData' })
-                },
-                [ErrorStates.UNKNOWN_ERROR]: {
-                    message: intl.formatMessage({ id: 'visualization.ErrorMessageGeneric' }),
-                    description: intl.formatMessage({ id: 'visualization.ErrorDescriptionGeneric' })
-                }
-            };
-            const errorProps = errorMap[errorMap.hasOwnProperty(error) ? error : ErrorStates.UNKNOWN_ERROR];
+            const errorProps = this.errorMap[this.errorMap.hasOwnProperty(error) ? error : ErrorStates.UNKNOWN_ERROR];
```

Comment:
Some linter in one of our project complains about calling Object.prototype functions directly on hashes. IMHO, this is a bit stupid. You would have to call Object.prototype.hasOwnProperty.call(this.errorMap, error);
The reason is, you can have a hash like this:
{
  hasOwnProperty: true
}
Is this a valid concern or should we disable this lint rule next time we stumble upon it?